Makefile

```diff
@@ -50,9 +50,17 @@ book: ## Builds the book & serves documentation site
 
 # --- testing -------------------------------------------------------------------------------------
 
+.PHONY: test-build
+test-build: ## Build the test binary
+	cargo nextest run --cargo-profile test-release --features concurrent,testing --no-run
+
```

Comment:
Before the debug assertions realization in miden-base PR 1230 the main win was to switch to:

```toml
[profile.test-release]
inherits = "dev"
opt-level = 1
codegen-units = 16 # I think this is not essential for the differences
```

The test execution time was roughly the same while the build time is significantly faster. So in particular, release doesn't seem to speed up the debug assertion in winter-prover. 
I think this suggests that any higher opt-level than 1 for winter prover would not really speed up that check, though you could try to selectively enable it and see if you find any difference in execution time - just to make sure.